kview/src/main/java/dev/ikm/komet/kview/mvvm/view/genediting/GenEditingDetailsController.java

```diff
@@ -358,17 +359,20 @@ private void setupSemanticDetailsUI(Latest<SemanticEntityVersion> semanticEntity
             // substitute each data type.
             if (dataTypeNid == TinkarTerm.COMPONENT_FIELD.nid()) {
                 // load a read-only component
-                ObservableField<EntityProxy> observableFields = obtainObservableField(getViewProperties(), semanticEntityVersionLatest, fieldRecord);
+                ObservableField<EntityProxy> observableField = obtainObservableField(getViewProperties(), semanticEntityVersionLatest, fieldRecord);
                 KlComponentFieldFactory klComponentFieldFactory = new KlComponentFieldFactory();
-                readOnlyNode = klComponentFieldFactory.create(observableFields, getViewProperties().nodeView(), false).klWidget();
+                readOnlyNode = klComponentFieldFactory.create(observableField, getViewProperties().nodeView(), false).klWidget();
+                observableFields.add(observableField);
             } else if (dataTypeNid == TinkarTerm.STRING_FIELD.nid() || fieldRecord.dataType().nid() == TinkarTerm.STRING.nid()) {
                 ObservableField<String> observableField = obtainObservableField(getViewProperties(), semanticEntityVersionLatest, fieldRecord);
                 KlStringFieldFactory klStringFieldFactory = new KlStringFieldFactory();
                 readOnlyNode = klStringFieldFactory.create(observableField, getViewProperties().nodeView(), false).klWidget();
                 observableFields.add(observableField);
             } else if (dataTypeNid == TinkarTerm.COMPONENT_ID_SET_FIELD.nid()) {
-                readOnlyNode = rowf.createReadOnlyComponentSet(getViewProperties(), fieldRecord);
-                observableFields.add(null);
+                ObservableField<Set<EntityProxy>> observableField = obtainObservableField(getViewProperties(), semanticEntityVersionLatest, fieldRecord);
```

Comment:
Can we obtain the observable field as `ObservableField<IntIdSet> observableField `?
We will probably need to modify the way drag and drop from outside concepts.

Reply:
Also, we will work together to integrate @bhharsh13 's PR related to refactoring this large conditional code related to `observableFields.add(observableField)` method call.